src/hooks/useForceInclusionHistory.ts

```diff
@@ -0,0 +1,186 @@
+        const l2TxHash = l2TxHashes[0];
+
+        let l2Status: TransactionStatus | undefined;
+
+        if (l2TxHash) {
+          try {
+            const l2Receipt = await l2Client.getTransactionReceipt({
+              hash: l2TxHash as `0x${string}`,
+            });
+            l2Status =
+              l2Receipt.status === 'success'
+                ? TransactionStatus.SUCCEEDED
+                : TransactionStatus.FAILED;
+          } catch (e) {
+            l2Status = TransactionStatus.PENDING;
+          }
+        }
+
+        return {
+          type: HistoryEntryType.L1_DEPOSIT,
+          txHash: event.transactionHash,
+          timestamp: Number(block.timestamp),
+          blockNumber: Number(block.number),
+          status: TransactionStatus.SUCCEEDED,
+          from: event.args.from,
+          to: event.args.to,
+          l2TransactionHash: l2TxHash,
+          l2Status,
+          l2Chain: chainConfigs[chainId].chain,
+          l1TxFee: receipt.gasUsed * receipt.effectiveGasPrice,
+        };
+      })
+    );
+
+    results.push(...batchResults);
+
+    if (i + batchSize < events.length) {
+      await sleep(delayMs);
+    }
+  }
+
+  return results;
+};
+
+export function useForceInclusionHistory(address: string) {
+  const [histories, setHistories] = useState<L1DepositHistory[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    async function fetchHistory() {
+      if (!address || !isAddress(address)) {
+        setIsLoading(false);
+        return;
+      }
+
+      try {
+        setIsLoading(true);
+        setError(null);
+
+        const l1RpcUrl = process.env.NEXT_PUBLIC_L1_RPC_URL;
+        if (!l1RpcUrl) {
+          throw new Error('L1 RPC URL not configured');
+        }
+
+        const l1Client = createPublicClient({
+          chain: sepolia,
+          transport: http(l1RpcUrl),
+        });
+
+        const DEPOSIT_EVENT = parseAbiItem(
+          'event TransactionDeposited(address indexed from, address indexed to, uint256 indexed version, bytes opaqueData)'
+        );
+
+        const allHistories: L1DepositHistory[] = [];
+
+        // Filter for OP Stack chains only
+        const opStackChains = Object.entries(chainConfigs).filter(
+          ([, config]) => config.isOpstack
+        );
+
+        // Process each chain sequentially to avoid too many concurrent requests
+        for (const [chainId, config] of opStackChains) {
+          const l2Client = createPublicClient({
+            chain: config.chain,
+            transport: http(),
+          });
+
+          const events = await l1Client.getLogs({
+            address: config.portalAddress,
+            event: DEPOSIT_EVENT,
+            args: {
+              from: address,
+            },
+            fromBlock: BigInt(config.startBlock),
+            toBlock: 'latest',
+          });
+
```

Comment:
The event query block range limit seems to be 5000 so we should probably just check the latest 5000 blocks (as `startBlock` tends to be way back in the past)?
```suggestion
          const EVENT_QUERY_BLOCK_RANGE = 5000 - 100; // Block range limit is 5000. We reduce it further by 100 as a buffer
          ...

          const latestBlockNumber = l1Client.getBlockNumber();
          const events = await l1Client.getLogs({
            address: config.portalAddress,
            event: DEPOSIT_EVENT,
            args: {
              from: address,
            },
            fromBlock: BigInt(latestBlockNumber - EVENT_QUERY_BLOCK_RANGE),
            toBlock: 'latest',
          });

```

Reply:
updated, also added comment on the history page to show that we only fetch recent histories